### PR TITLE
Initial onboarding for PIM for Azure RBAC API

### PR DESCRIPTION
**Pull request by wn9237**
initial documents for pim for azure rbac

### CODE REVIEW

**Comment by jasonjoh** [comment]
@wn9237 Looks like you're having trouble with AppVeyor checks. Do you still need to merge this pull request?

**Comment by wn9237** [comment]
Hi @jasonjoh,
we have resolved the app veyor issues. please review and let me know whether there is more issues.

**Comment by jasonjoh** [comment]
Great! Adding @Lauragra to double-check this. Laura, do we want to redirect this to Staging instead of a direct master pull?

**Comment by Lauragra** [comment]
Yes, please target the Staging branch. Since this is new content, please assign me as a reviewer and send me your proposed TOC structure for this content. Are the APIs live now, or if not, what is the ETA for the APIs/docs going live?

Thanks,
Laura

**Comment by jasonjoh** [comment]
@Lauragra: I changed the target to Staging and added you as a reviewer.

@wn9237: Please send Laura your proposed table of contents structure for these topics.

File: api-reference/beta/api/governanceroleassignmentrequest_post.md [hunk 1]
```diff
@@ -0,0 +1,549 @@
+# Post governanceRoleAssignmentRequest
```
Comment (Lauragra) [inline api-reference/beta/api/governanceroleassignmentrequest_post.md]: Please use "Create" in the title.

File: api-reference/beta/api/governanceroleassignmentrequest_post.md [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Create a role assignment request to represent the operation you want on a role assignment. The following table lists the operations.
+
+| Operation       | Type | 
+|:---------------|:----------|
```
Comment (Lauragra) [inline api-reference/beta/api/governanceroleassignmentrequest_post.md]: Please fix the table - looks like some rows have three columns; add the third column to the header. Is it by design that the column is blank for some operations?

File: api-reference/beta/api/governanceroledefinition_get.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+| Authorization  | Bearer {code}|
+
+
+## Request body
+Do not supply a request body for this method.
+## Response
+If successful, this method returns a `200 OK` response code and [governanceRoleDefinition](../resources/governanceroledefinition.md) object in the response body.
+## Example
+Get details of role Definition "DNS Zone Contributor" in subscription "Wingtip Toys - Prod"
+<!-- {
```
Comment (Lauragra) [inline api-reference/beta/api/governanceroledefinition_get.md]: Please don't use quotation marks for role names and lowercase the word Definition. Make sure that the introduction is a complete sentence that ends with a period.

File: api-reference/beta/api/governanceroleassignmentrequest_post.md (hunk 1 above)
Comment (wn9237) [reply api-reference/beta/api/governanceroleassignmentrequest_post.md]: Sure. our PM is working on a TOC for you, will update here once it is done.

**Comment by wn9237** [comment]
@Lauragra I have updated the  doc where you have commented. Please take a look and move forward. Thanks.

File: api-reference/beta/api/governanceroledefinition_get.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+Besides the permission scope, this API requires the requestor to have at least one role assignment on the resource, which the [governanceRoleDefinition](../resources/governanceroledefinition.md) belongs to.
+
+## HTTP request
+<!-- { "blockType": "ignored" } -->
+```http
+GET /privilegedAccess/azureResources/resources/{resourceId}/roleDefinitions/{id}
+GET /privilegedAccess/azureResources/roleDefinitions/{id}?$filter=resourceId+eq+'{resourceId}'
+```
+## Optional query parameters
+This method does **not** support the [OData Query Parameters](http://graph.microsoft.io/docs/overview/query_parameters) to help customize the response.
```
Comment (Lauragra) [inline api-reference/beta/api/governanceroledefinition_get.md]: Global: The URL for the Query parameters topic is out of date. Please change to a relative link: ../../../concepts/query_parameters.md.

File: api-reference/beta/api/governancerolesetting_list.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
+# List governanceRoleSettings
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Retrieve a collection of [governanceRoleSetting](../resources/governancerolesetting.md) on a resource.
```
Comment (Lauragra) [inline api-reference/beta/api/governancerolesetting_list.md]: In list topics, be sure to match the plurality.

File: api-reference/beta/resources/governancepermission.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# `permission` type
```
Comment (Lauragra) [inline api-reference/beta/resources/governancepermission.md]: Remove the tick marks and update the title to "permission resource type".

File: api-reference/beta/resources/governanceresource.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
+# governanceResource
+
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceresource.md]: Global: Include the "resource type" descriptor in the title.

File: api-reference/beta/resources/governanceresource.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents resources that could be managed by Privileged Identity Management (PIM). 
+* For scope of Azure resources, the resource can be a subscription, a resource group, and a resource such as a virtual machine, a SQL database, etc.
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceresource.md]: Please clarify the phrase "For scope of Azure resources". Did you intend to start a new paragraph there or is it a note or bullet?

File: api-reference/beta/resources/governancepermission.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+Represents the access permission that a `subject` has to a specific `resource`.  
+
+
+
+### Properties
```
Comment (Lauragra) [inline api-reference/beta/resources/governancepermission.md]: Global: In the resource topics, change the H3s to H2s (##).

File: api-reference/beta/resources/governanceresource.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+|:---------------|:--------|:----------|
+|[List](../api/governanceresource_list.md) | [governanceResource](../resources/governanceresource.md) collection|List a collection of resources the requestor has access to.|
+|[Get](../api/governanceresource_get.md) | [governanceResource](../resources/governanceresource.md) |Read properties and relationships of a resource entity specified by id.|
+
+No `POST`, `PUT`, `PATCH`, `DELETE` are supported on `resources` entity set for now.
+
+### Properties
+| Key | Property	        |Type	      |Description|
+|:----|:------------------|:----------|:----------|
+|✓    |id                 |String     |The id of the resource. It is in GUID format.|
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceresource.md]: What does the Key column with one check mark mean in this context?

File: api-reference/beta/resources/governanceroleassignmentrequeststatus.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
+# governanceRoleAssignmentRequestStatus
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents the status of the [governanceRoleAssignmentRequest](../resources/governanceroleassignmentrequest.md).
+
+
+### Properties
+Property 	   | Type |Description|
+|:----|:-------------|:-----|:-------|:----------|
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceroleassignmentrequeststatus.md]: Please ensure that the tables are formatted correctly. This table has three columns so make sure that you have three columns only in each row.

File: api-reference/beta/resources/governanceroledefinition.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
+# governanceRoleDefinition 
+Represents the role definitions.
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceroledefinition.md]: Please move the beta notation to come before the summary.

File: api-reference/beta/resources/governanceroledefinition.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
+# governanceRoleDefinition 
+Represents the role definitions. 
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+* For scope of Azure resources, it can represent Azure RBAC roles, such as Owner, Reader, Contributor, etc.
+
```
Comment (Lauragra) [inline api-reference/beta/resources/governanceroledefinition.md]: Clarify the "For scope..." phrase and formatting.

File: api-reference/beta/resources/governancerolesetting.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
+# governanceRoleSetting
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents a set of configurations on each role definition that needs to be evaluated against when role assignments are created or modified. For example, role settings might include "maximum assignment duration", "MFA required on activation", and so on.
+
+### Methods
+
+| Method		  | Return Type	|Description|
+|:---------------|:--------|:--------|:----------|
```
Comment (Lauragra) [inline api-reference/beta/resources/governancerolesetting.md]: Fix table.

**Comment by dkershaw10** [comment]
@wn9237 Please work with @lleonard-msft or @Lauragra to create a ToC for all this material (under Identity and Access node in /beta) please. @wn9237 you also need to create a changelog.md update here too.

File: api-reference/beta/resources/governanceresource.md (hunk 10 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceresource.md]: it means the unique identitfier of the entity.

File: api-reference/beta/api/governanceroledefinition_get.md (hunk 4 above)
Comment (wn9237) [reply api-reference/beta/api/governanceroledefinition_get.md]: The link has been updated globally.

File: api-reference/beta/resources/governancepermission.md (hunk 6 above)
Comment (wn9237) [reply api-reference/beta/resources/governancepermission.md]: updated globally.

File: api-reference/beta/resources/governanceresource.md (hunk 7 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceresource.md]: Done globally.

File: api-reference/beta/resources/governanceresource.md (hunk 8 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceresource.md]: kind of revised the wording here.

File: api-reference/beta/resources/governancepermission.md (hunk 9 above)
Comment (wn9237) [reply api-reference/beta/resources/governancepermission.md]: updated globally.

File: api-reference/beta/resources/governanceroleassignmentrequeststatus.md (hunk 11 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceroleassignmentrequeststatus.md]: corrected the table format.

File: api-reference/beta/resources/governanceroledefinition.md (hunk 12 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceroledefinition.md]: done.

File: api-reference/beta/resources/governanceroledefinition.md (hunk 13 above)
Comment (wn9237) [reply api-reference/beta/resources/governanceroledefinition.md]: done

File: api-reference/beta/resources/governancerolesetting.md (hunk 14 above)
Comment (wn9237) [reply api-reference/beta/resources/governancerolesetting.md]: fixed.

File: api-reference/beta/resources/governancerulesetting.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
+# governanceRuleSetting
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents the rules that compose the role settings. It is the type of each rule to evaluate role assignments.
```
[the remark this answers is not in the document]
Comment (wn9237) [reply api-reference/beta/resources/governancerulesetting.md]: done. also removed the second sentence.

File: api-reference/beta/resources/governanceschedule.md [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+Represents the schedule for a [governanceRoleAssignmentRequest](../resources/governanceroleassignmentrequest.md) . For a role assignment request, schedule controls when to perform the role assignment operation, when to stop the role assignment, and how frequent to do the role assignment operation for the recurrent scenario. 
+
+
+
+### Properties
+| Property	   | Type	|Description|
+|:---------------|:--------|:----------|
+|startDateTime|DateTimeOffset|The start time of the role assignment. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`|
+|endDateTime|DateTimeOffset|The end time of the role assignment. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`. *Note: if the value is `null`, it indicates a permanent assignment.*|
+|type|String|The role assignment schedule type. The value can be ``Once``. (``Daily`` ,``Weekly`` and other types are coming soon)|
```
[the remark this answers is not in the document]
Comment (wn9237) [reply api-reference/beta/resources/governanceschedule.md]: yes. revised the description here.

**Comment by wn9237** [comment]
@Lauragra Let me know if you have any more comments/suggestions. If it looks good to you,let me know what is the next step.

File: api-reference/beta/resources/governanceroleassignmentrequest.md [hunk 17]
```diff
@@ -17,21 +17,21 @@ Represents the request for role assignment operations in Privilegd Identity Mana
 |[Update](../api/governanceroleassignmentrequest_update.md)| [governanceRoleAssignmentRequest](../resources/governanceroleassignmentrequest.md)|Administrators update the decisions on requests if the requests are in status of `PendingAdminDecision`.|
 
 ## Properties
-| Key | Property	                | Type	        |Required|Description|
-|:----|:--------------------------|:--------------|:-------|:----------|
-|✓    |id                         |String         |✓      |The id of the role assignment request.|
-|     |resourceId                 |String         |✓       |The id of the resource which the role assignment request is associated with.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/governanceroleassignmentrequest.md]: You should probably now indicate in the description, which of the properties are required.

Comment (wn9237) [reply api-reference/beta/resources/governanceroleassignmentrequest.md]: Valid point. will add that.

**Pull request merged**